astar_path_planner/src/astar_path_planner_plugin.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright 2021 RoboJackets
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include <astar_path_planner/astar_path_planner.hpp>
+#include <nav2_core/global_planner.hpp>
+#include <pluginlib/class_list_macros.hpp>
+#include <memory>
+#include <string>
+
+namespace astar_path_planner
+{
+
+class AStarPathPlannerPlugin : public nav2_core::GlobalPlanner
+{
+public:
+  void configure(
+    rclcpp_lifecycle::LifecycleNode::SharedPtr node,
+    std::string name, std::shared_ptr<tf2_ros::Buffer> tf,
+    std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmap_ros) override
+  {
+    node_ = node;
+    global_frame_ = costmap_ros->getGlobalFrameID();
+    costmap_ros_ = costmap_ros;
+    AStarPathPlanner::DeclareParameters(node);
+  }
+
+  void cleanup() override {}
+
+  void activate() override {}
+
+  void deactivate() override {}
+
+  nav_msgs::msg::Path createPlan(
+    const geometry_msgs::msg::PoseStamped & start,
+    const geometry_msgs::msg::PoseStamped & goal) override
+  {
+    nav_msgs::msg::Path path;
+    path.header.stamp = node_->now();
```

Comment:
Is there a way to make this time match the pose time? I'm not sure if start is the current position or not.

Reply:
Start is _usually_ the current pose (though that's really up to the caller of the service).

My understanding is that the path header stamp is expected to represent when the path was generated, not necessarily tied to the timestamps of the input poses.